<?xml version="1.0"?>
<robot name="ThreeLinkRobot">
  <link name="body1">
    <visual>
      <geometry>
        <box size = "0.623 0.063 0.191"/>
      </geometry>
    </visual>
    <inertial>
      <origin xyz="0.0064116 -3.8482E-05 6.9037E-05" rpy="0 0 0" />
      <mass value="0.5476" />
      <inertia
        ixx="0.0010352"
        ixy="6.3703E-06"
        ixz="0.00012413"
        iyy="0.0062106"
        iyz="-6.3903E-06"
        izz="0.0070685" />
    </inertial>
    <collision>
      <origin xyz="-.095 0 0" rpy="0 0 0" />
      <geometry>
          <box size = "0.623 0.063 0.191"/>
      </geometry>
    </collision>
  </link>

  <link name="body2">
    <visual>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <geometry>
        <box size="0.6 0.2 0.1"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <mass value="0.5476" />
      <inertia
        ixx="0.0010352"
        ixy="6.3703E-06"
        ixz="0.00012413"
        iyy="0.0062106"
        iyz="-6.3903E-06"
        izz="0.0070685" />
    </inertial>
    <collision>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <geometry>
          <box size="0.6 0.2 0.1"/>
      </geometry>
    </collision>
  </link>

  <link name="body3">
    <visual>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <geometry>
        <box size="0.6 0.2 0.1"/>
      </geometry>
    </visual>
    <inertial>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <mass value="0.5476" />
      <inertia
        ixx="0.0010352"
        ixy="6.3703E-06"
        ixz="0.00012413"
        iyy="0.0062106"
        iyz="-6.3903E-06"
        izz="0.0070685" />
    </inertial>
    <collision>
      <origin rpy="0 1.57075 0" xyz="0 0 -0.3"/>
      <geometry>
          <box size="0.6 0.2 0.1"/>
      </geometry>
    </collision>
  </link>

  <joint name="joint1" type="revolute">
    <parent link="body1"/>
    <child link="body2"/>
    <origin xyz="0.22 0 0.25"/>
    <limit lower="-0.65" upper="0.65" effort="100" velocity="18" />
  </joint>

  <joint name="joint2" type="revolute">
    <parent link="body2"/>
    <child link="body3"/>
    <origin xyz="1.0 2.0 3.0"/>
    <limit lower="-1.0" upper="2.0" effort="10" velocity="18" />
  </joint>

  <force_element name="propeller_force">
    <propellor name="propeller" lower_limit="0" upper_limit="100"
    	         scale_factor_thrust="0.05159" scale_factor_moment="0.005"
    	         visual_geometry="true" visual_diameter="0.2">
      <parent link="body1"/>
      <origin
        xyz="0 0 0"
        rpy="0 0 0" />
      <axis xyz="1 0 0" />
    </propellor>
  </force_element>

  <force_element name="my_spring">
    <linear_spring_damper rest_length=".323342" stiffness="10000" damping="100">
      <link1 link="body2" xyz="0 -0.801993 .323342"/>
      <link2 link="body3"/>
    </linear_spring_damper>
  </force_element>
</robot>